vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+)
+
+ibor_non_icripd <- event(
+  description = paste(
+    "Define NON-iCR/NON-iUPD for best overall response (iBOR) as NON-iCR/NON-iUPD",
+    "occuring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC == "NON-iCR/NON-iUPD" & ADT >= RANDDT + 42,
+  set_values_to = exprs(AVALC = "NON-iCR/NON-iUPD")
+)
+
+ibor_icpd <- event(
+  description = "Define progressive disease (iCPD) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iCPD",
+  set_values_to = exprs(AVALC = "iCPD")
+)
+
+ibor_iupd <- event(
+  description = "Define progressive disease (iUPD) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iUPD",
+  set_values_to = exprs(AVALC = "iUPD")
+)
+
+ibor_ne <- event(
+  description = paste(
+    "Define not evaluable (NE) for best overall response (iBOR) as iSD, NON-iCR/NON-iUPD,",
+    "or NE (should be specified after ibor_isd and ibor_non_icripd)"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("iSD", "NON-iCR/NON-iUPD", "NE"),
+  set_values_to = exprs(AVALC = "NE")
+)
+
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(ibor_icr, ibor_ipr, ibor_isd, ibor_non_icripd, ibor_icpd, ibor_iupd, ibor_ne, no_data_missing),
+    set_values_to = exprs(
+      PARAMCD = "IBOR",
+      PARAM = "Immune Best Overall Response by Investigator (confirmation not required)",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRecist",
+      AVAL = aval_resp_new(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, RANDDT, ANL01FL),
+  filter = PARAMCD == "IBOR"
+)
+```
+
+## Derive Response Parameters requiring Confirmation {#confirm}
+
+Any of the above response parameters can be repeated for "confirmed" responses
+only. For these the function `admiral::derive_extreme_event()` can be used with
+different events. Some of the other functions from above can then be re-used
+passing in these confirmed response records. See the examples below of derived
+parameters requiring confirmation. The assessment and the confirmatory
+assessment here need to occur at least 28 days apart *(without any +1 applied to
+this calculation of days between visits)*, using the `icrsp_y_cr`[^1],
+`icrsp_y_ipr`[^2], `icbor_icr`[^3], and `icbor_ipr`[^4] event.
+
+[^1]:
+```{r}
+icrsp_y_icr <- event_joined(
+  description = paste(
+    "Define confirmed response as iCR followed by CR at least 28 days later and",
+    "at most one NE in between"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  order = exprs(ADT),
+  first_cond = AVALC.join == "CR" &
+    ADT.join >= ADT + days(28),
```

Comment:
I would use confirmation_period instead of 28 days fixed, as it was done in adrs.html (in all 4 definitions of events below)

Reply:
will update